### PR TITLE
Suppress openssl-3.0 warnings

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/resolve/resolved-dns-dnssec.c [hunk 1]
```diff
@@ -13,6 +13,14 @@
 #include "sort-util.h"
 #include "string-table.h"
 
+#if HAVE_OPENSSL
+#  pragma GCC diagnostic push
+#    pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+DEFINE_TRIVIAL_CLEANUP_FUNC_FULL(RSA*, RSA_free, NULL);
```
Comment (bluca) [inline src/resolve/resolved-dns-dnssec.c:19]: Any specific reason to move these here instead of adding the pragma to the header where they were before?

**Comment by bluca** [comment]
Clang is sad:
```
../src/resolve/resolved-dns-dnssec.c:19:1: error: unused function 'RSA_freep' [-Werror,-Wunused-function]
DEFINE_TRIVIAL_CLEANUP_FUNC_FULL(RSA*, RSA_free, NULL);
^
../src/basic/macro.h:378:28: note: expanded from macro 'DEFINE_TRIVIAL_CLEANUP_FUNC_FULL'
        static inline void func##p(type *p) {                   \
                           ^
<scratch space>:102:1: note: expanded from here
RSA_freep
^
../src/resolve/resolved-dns-dnssec.c:20:1: error: unused function 'EC_KEY_freep' [-Werror,-Wunused-function]
DEFINE_TRIVIAL_CLEANUP_FUNC_FULL(EC_KEY*, EC_KEY_free, NULL);
^
../src/basic/macro.h:378:28: note: expanded from macro 'DEFINE_TRIVIAL_CLEANUP_FUNC_FULL'
        static inline void func##p(type *p) {                   \
                           ^
<scratch space>:103:1: note: expanded from here
EC_KEY_freep
^
2 errors generated.
```

File: src/resolve/resolved-dns-dnssec.c [hunk 2]
```diff
@@ -13,6 +13,14 @@
 #include "sort-util.h"
 #include "string-table.h"
 
+#if HAVE_OPENSSL
```
Comment (yuwata) [inline src/resolve/resolved-dns-dnssec.c]: `#if PREFER_OPENSSL`??

File: src/resolve/resolved-dns-dnssec.c (hunk 1 above)
Comment (keszybz) [reply src/resolve/resolved-dns-dnssec.c:19]: I should have mentioned that in the commit message. I wanted to do that in a separate step but then just did one commit… Those functions are only used in this file, and I don't think we should add any more uses. So it seems better to "localize" the ugly parts in that one file.

Comment (yuwata) [reply src/resolve/resolved-dns-dnssec.c:19]: These are used only in this file, and should not be used in newer code. I think moving them there is reasonable.

File: src/resolve/resolved-dns-dnssec.c (hunk 2 above)
Comment (keszybz) [reply src/resolve/resolved-dns-dnssec.c]: Oh right. Thanks.

File: src/resolve/resolved-dns-dnssec.c (hunk 1 above)
Comment (bluca) [reply src/resolve/resolved-dns-dnssec.c:19]: Suspected as much, but wanted to double check

**Comment by keszybz** [comment]
Should be better now.

**Review by yuwata — APPROVED** [review]
LGTM.

**Comment by keszybz** [comment]
centos ci (arch linux): TEST-64-UDEV-STORAGE failed.

**Comment by keszybz** [comment]
It seems to compile fine… The new code is not used the ubuntuautopkg builds, so I don't think we need to wait for those.

**Pull request merged**